src/app/zap-templates/helper-chip.js

```diff
@@ -69,6 +72,35 @@ function isCommandAvailable(clusterSide, incoming, outgoing, source, name) {
   return false;
 }
 
+/**
+ * Returns CHIP specific type for ZCL framework
+ * This function is flawed since it relies on the
+ * type label for CHIP type conversion. CHIP specific XML should have the
+ * correct type directly embedded inside.
+ *
+ * @param {*} type : The xml type to be converted
+ */
+function asChipUnderlyingType(label, type) {
+
+  if (zclHelper.isStrEqual(label, "endpoint")) {
+    return 'chip::EndpointId'
+  } else if (zclHelper.isStrEqual(label, "endpointId")) {
+    return 'chip::EndpointId'
+  } else if (zclHelper.isStrEqual(label, "CLUSTER_ID")) {
+    return 'chip::ClusterId'
+  } else if (zclHelper.isStrEqual(label, "ATTRIBUTE_ID")){
+    return 'chip::AttributeId'
+  } else if (zclHelper.isStrEqual(label, "groupId")) {
+    return 'chip::GroupId'
+  } else if (zclHelper.isStrEqual(label, "commandId")) {
+    return 'chip::CommandId'
+  } else {
+    // Might want to use asZclUnderlyingType instead. TBD
```

Comment:
The answer to this is yes at the moment. Can you use that instead ?

Reply:
For now I'll stick with the `asUnderlyingType` since the `asUnderlyingZclType` seems to have a problem depending on the context it is used in.